src/backend/libpq/rangerrest.c

```diff
@@ -49,10 +49,26 @@ char* AclObjectKindStr[] =
 	"none"               /* MUST BE LAST */
 };
 
+static int request_id = 1;
+
+void getClientIP(char *remote_host)
+{
+	Port *port = MyProcPort;
+	char remote_port[32];
+	remote_port[0] = '\0';
+
+	int ret = pg_getnameinfo_all(&port->raddr.addr, port->raddr.salen,
+					remote_host, sizeof(remote_host),
```

Comment:
There may be wrong:
sizeof(remote_host) always == sizeof(void *) == 8byte

Should define a constant for buffer size.